src/app/mytools/sidebar-accordion/sidebar-accordion.component.ts

```diff
@@ -1,24 +1,97 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnInit, SimpleChanges } from '@angular/core';
 import { DockstoreTool } from 'app/shared/swagger';
 import { Observable } from 'rxjs';
 import { ToolQuery } from '../../shared/tool/tool.query';
 import { OrgToolObject } from '../my-tool/my-tool.component';
 
+interface groupEntriesBySource {
+  groupEntryInfo: OrgToolObject<DockstoreTool>[];
+  sourceControlTitle: string;
+}
+
 @Component({
   selector: 'app-sidebar-accordion',
   templateUrl: './sidebar-accordion.component.html',
   styleUrls: ['./sidebar-accordion.component.scss'],
 })
 export class SidebarAccordionComponent implements OnInit {
-  constructor(private toolQuery: ToolQuery) {}
   @Input() openOneAtATime;
   @Input() groupEntriesObject: OrgToolObject<DockstoreTool>[];
   @Input() refreshMessage;
-
   public toolId$: Observable<number>;
   activeTab = 0;
 
-  ngOnInit(): void {
+  public sortedTools: groupEntriesBySource[] = [
```

Comment:
Looks good overall!  Something to consider...

You might use a `Map` for `sortedTools` rather than an array.  When you sort, the map key would be the value of `sourceControl` (normalized, perhaps by stripped of the leading 'registry.' etc) and the map value would be the list of tools.  This would avoid the subsequent big chain of `if`s, need to match up source control values to array indexes, etc.  Then, in your template, you could iterate through the source control values (keys) of interest, in whatever order you wanted, and retrieve the lists of tools by simply getting the value from the map.

If you did this, `sortedTools` probably gets renamed to `sourceControlToTools` or something like that.

Reply:
Thanks for the suggestion Steve! I updated `sortedTools` and `sortedWorkflows` to use a Map instead of an array. Instead of having the array of tools as the `value` I used the `groupEntriesBySource` interface from the original array so that I could still keep the `sourceControlTitle` (since this is used as the display name). I removed the large block of `if else` statements  for Workflows and there is now only one `if else` for Tools (to handle additional tools that do not match with any of the existing keys). Much happier with the code now, this was my first time learning about Maps, thank you!